### ISSUE TITLE
reduce `resample` ringing?

### ISSUE THREAD

**Issue by ssfrr** [JuliaDSP/DSP.jl#137]
Currently `resample` zero-pads the input and shifts the output so that the input and output line up better. The zero-padding introduces a discontinuity that can cause substantial ringing:

```julia
using DSP
using PlotlyJS

# create smooth signal and cut off filter ramp-up
x1 = filt(digitalfilter(Lowpass(0.04), Butterworth(2)), rand(100))[30:end]

x2 = resample(x1, 99//100)
t1 = (0:(length(x1)-1))/100
t2 = (0:(length(x2)-1))/99
plot([
    scatter(x=t1, y=x1),
    scatter(x=t2, y=x2)
])
```

Would it be a problem to extend out the last value rather than zero-padding to reduce the ringing?

**Comment by staticfloat** [comment]
Extending the last value is good, mirroring the first few samples of the input is also a common workaround.

**Comment by ssfrr** [comment]
closed in favor of #241
